test/src/test/java/hudson/util/ProcessTreeKillerTest.java

```diff
@@ -135,7 +134,10 @@ public void doNotKillProcessWithCookie() throws Exception {
         public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
             EnvVars envvars = build.getEnvironment(listener);
             envvars.addLine("BUILD_ID=dontKillMe");
-            proc = launcher.launch().envs(envvars).cmds("nohup", "sleep", "100000").start();
+            final String[] cmd = Functions.isWindows()
+                    ? new String[]{"ping", "-n", "100000", "localhost"}
```

Comment:
perhaps `timeout /t 99999 /nobreak` is more appropriate

Reply:
`timeout` exits when build finishes, no idea why. `ping` doesn't.